--- ./work/edit_transaction.work.js	2026-05-03 01:16:14.733244400 +0530
+++ ./work/edit_transaction.work2.js	2026-05-03 01:17:54.894606900 +0530
@@ -7215,11 +7215,13 @@
                                 console.error(
                                     "window.expectedSourceTypes is unexpectedly empty.",
                                 );
+                            let parsedDate = moment.parseZone(e.date);
                             var a = {
                                 transaction_journal_id:
                                     e.transaction_journal_id,
                                 description: e.description,
-                                date: e.date.substring(0, 16),
+                                dateOffsetTarget: parsedDate.utcOffset(),
+                                date: parsedDate.local().format().substring(0, 16),
                                 reconciled: e.reconciled,
                                 amount: this.roundNumber(
                                     this.positiveAmount(e.amount),
@@ -7411,6 +7413,7 @@
                             (e.currency_id = a),
                             (c = e.date),
                             t > 0 && (c = this.transactions[0].date),
+                            (c = moment(c).utcOffset(this.transactions[0].dateOffsetTarget).format().substring(0, 16)),
                             "withdrawal" === n &&
                                 "" === l &&
                                 (s = window.cashAccountId),
